keras/preprocessing/image.py

```diff
@@ -249,8 +249,8 @@ def reset(self):
 
     def _flow_index(self, N, batch_size=32, shuffle=False, seed=None):
         while 1:
-            index_array = np.arange(N)
```

Comment:
`index_array` is ALWAYS initialized when `_flow_index`, so some indexes may not be used 😢